packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Models/Types/SystemModelSerializationTypeProvider.cs

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Microsoft.Generator.CSharp.Expressions;
+
+namespace Microsoft.Generator.CSharp.ClientModel
+{
+    /// <summary>
+    /// This class provides the set of serialization models, methods, and interfaces for a given model.
+    /// </summary>
+    internal sealed class SystemModelSerializationTypeProvider : TypeProvider
+    {
+        internal static readonly Parameter SerializationOptionsParameter =
+            new("options", null, typeof(ModelReaderWriterOptions), null, ValidationType.None, null);
+        private readonly CSharpType _iJsonModelTInterface;
+        private readonly CSharpType? _iJsonModelObjectInterface;
+        private readonly CSharpType _iPersistableModelTInterface;
+        private readonly CSharpType? _iPersistableModelObjectInterface;
+
+        public SystemModelSerializationTypeProvider(ModelTypeProvider model) : base(null)
+        {
+            Model = model;
+            Name = model.Name;
+            Json = BuildJsonSerialization();
+            // Initialize the serialization interfaces
+            _iJsonModelTInterface = new CSharpType(typeof(IJsonModel<>), Model.Type);
+            _iJsonModelObjectInterface = Model.IsStruct ? (CSharpType)typeof(IJsonModel<object>) : null;
+            _iPersistableModelTInterface = new CSharpType(typeof(IPersistableModel<>), Model.Type);
+            _iPersistableModelObjectInterface = Model.IsStruct ? (CSharpType)typeof(IPersistableModel<object>) : null;
+        }
+
+        public override string Name { get; }
+        public ModelTypeProvider Model { get; }
+        public JsonObjectSerialization? Json { get; }
+
+        /// <summary>
+        /// Builds the serialization methods for the model. If the serialization supports JSON, it will build the JSON serialization methods.
+        /// </summary>
+        /// <returns>A list of serialization and deserialization methods for the model.</returns>
+        protected override CSharpMethod[] BuildMethods()
+        {
+            var methods = new List<CSharpMethod>();
+
+            // Build the JSON serialization methods
+            if (Json != null)
```

Comment:
I think if this class in constructed we should assume that json serialization is wanted since this is what the class does.  The code that constructs this class should only do it if needed.